src/encryption/encryption.go

```diff
@@ -0,0 +1,103 @@
+package encryption
+
+import (
+	"keboola-as-code/src/state"
+	"keboola-as-code/src/utils"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/iancoleman/orderedmap"
+	"go.uber.org/zap"
+)
+
+func isKeyToEncrypt(key string) bool {
+	return strings.HasPrefix(key, "#")
+}
+
+func isEncrypted(value string) bool {
+	return regexp.MustCompile(`^KBC::(ProjectSecure|ComponentSecure|ConfigSecure)::.+$`).MatchString(value)
+}
+
+type Group struct {
+	object state.ObjectState // config or configRow state object
+	values []Value
+}
+
+type Value struct {
+	key     string // key e.g. "#myEncryptedValue"
+	value   string // value to encrypt
+	keyPath path   // path to value from config
+}
+
+type finder struct {
+	errors *utils.Error
+	values []Value
+}
+
+func (f *finder) parseArray(array []interface{}, keyPath path) {
+	for idx, value := range array {
+		f.parseConfigValue(strconv.Itoa(idx), value, append(keyPath, sliceStep(idx)))
+	}
+}
+
+func (f *finder) parseOrderedMap(content *orderedmap.OrderedMap, keyPath path) {
+	for _, key := range content.Keys() {
+		mapValue, _ := content.Get(key)
+		f.parseConfigValue(key, mapValue, append(keyPath, mapStep(key)))
+	}
+}
+
+func (f *finder) parseConfigValue(key string, configValue interface{}, keyPath path) {
+	switch value := configValue.(type) {
+	case orderedmap.OrderedMap:
+		f.parseOrderedMap(&value, keyPath)
+	case string:
+		if isKeyToEncrypt(key) && !isEncrypted(value) {
+			f.values = append(f.values, Value{key, value, keyPath})
+		}
+	case []interface{}:
+		f.parseArray(value, keyPath)
+	}
+}
+
+func (f *finder) FindValues(projectState *state.State) []Group {
+	var groups []Group
+	for _, object := range projectState.All() {
+		f.values = nil
```

Comment:
@kacurez Mam len jednu poznamku. 
Toto pouzitie values je funkcne, ale trocha divne, ... mozes rovno zapisovat do `Group.values`.
Aby som to nezdrzoval rovno to commitnem

Reply:
ok diky, tie errors zatial zmazem ked sa nepouzivaju

Reply:
ok, hod ma potom na review, a mozeme to mergnut